protocol/xmrmaker/swap_state_test.go

```diff
@@ -46,11 +45,8 @@ var (
 	defaultTimeoutDuration, _ = time.ParseDuration("86400s") // 1 day = 60s * 60min * 24hr
 )
 
-func newTestXMRMaker(t *testing.T) *Instance {
-	pk, err := ethcrypto.HexToECDSA(common.DefaultPrivKeyXMRMaker)
-	require.NoError(t, err)
-
-	ec, err := ethclient.Dial(common.DefaultEthEndpoint)
```

Comment:
curious why you changed the ethclient to be an argument instead of creating it here? seems like basically every test case creates one now instead and passes it in

Reply:
The ethclient is creating a persistent websocket connection.  To get it properly closed by the unit tests even if there is an error, we need to have a `defer ec.close()` statement at a scope that is far enough out to not close the connection too early.

I thought about creating a test suite to reduce the boiler plate code for creating the connection, but decided that it would be too big of a change to take on in this PR.  I can do it in a future PR if you think it will make the code easier to maintain.

Reply:
you can keep it in the `newTestXMRMaker` function and use `t.Cleanup()` within it to call `ec.Close()` when each unit test is done. that should have the same functionality I believe